src/util/tokenpipe.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#include <util/tokenpipe.h>
+
+#include <config/bitcoin-config.h>
+
+#ifndef WIN32
+
+#include <errno.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+std::optional<TokenPipe> TokenPipe::create()
```

Comment:
Doesn't matter, but net_processing uses `make` as the name for PeerManager's nearly-but-not-quite-a-constructor.

Reply:
Thanks, I'm fine with a different name, `create` is the convention in rust but Make is fine with me too.

Reply:
Yeah, either way. PeerManager could change too; it was just copied from some stackoverflow link or something.